boards/ti/lp_mspm0g3507/docs/index.rst

```diff
@@ -0,0 +1,132 @@
+high speed comparators with built-in reference DACs, two zero-drift zero-crossover op-amps with programmable
+gain, and one general-purpose amplifier. These devices also offer intelligent digital peripherals such as two
+16-bit advanced control timers, five general-purpose timers (with one 16-bit general-purpose timer for QEI
+interface, two 16-bit general-purpose timers for STANDBY mode, and one 32-bit general-purpose timer), two
+windowed-watchdog timers, and one RTC with alarm and calendar modes. These devices provide data integrity
+and encryption peripherals (AES, CRC, TRNG) and enhanced communication interfaces (four UART, two I2C,
+two SPI, CAN 2.0/FD).
+
+.. figure:: img/lp_mspm0g3507.png
+     :align: center
+     :alt: MSPM0G3507 LaunchPad development board
+
+Features:
+=========
+
+- Onboard XDS110 debug probe
+- EnergyTrace technology available for ultra-low-power debugging
+- 2 buttons, 1 LED and 1 RGB LED for user interaction
+- Temperature sensor circuit
+- Light sensor circuit
+- External OPA2365 (default buffer mode) for ADC (up to 4 Msps) evaluation
+- Onboard 32.768-kHz and 40-MHz crystals
+- RC filter for ADC input (unpopulated by default)
+
+Details on the MSPM0G3507 LaunchPad can be found on the `TI LP_MSPM0G3507 Product Page`_.
+
+Supported Features
+==================
+
+The MSPM0G3507 LaunchPad development board configuration supports the following hardware features:
+
++-----------+------------+-----------------------+
+| Interface | Controller | Driver/Component      |
++===========+============+=======================+
+| NVIC      | on-chip    | nested vectored       |
+|           |            | interrupt controller  |
++-----------+------------+-----------------------+
+| SYSTICK   | on-chip    | system clock          |
++-----------+------------+-----------------------+
+| UART      | on-chip    | serial                |
++-----------+------------+-----------------------+
+| GPIO      | on-chip    | gpio                  |
++-----------+------------+-----------------------+
+| PINMUX    | on-chip    | pinctrl               |
++-----------+------------+-----------------------+
+
+More details about the supported peripherals are available in `MSPM0G3507 TRM`_.
+Other hardware features are not currently supported by the Zephyr kernel.
+
+Building and Flashing
+*********************
+
+Building
+========
+
+Follow the :ref:`getting_started` instructions for Zephyr application development.
+
+For example, to build the :ref:`hello_world` application for the MSPM0G3507 LaunchPad:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: lp_mspm0g3507
+   :goals: build
+
+The resulting ``zephyr.bin`` binary in the build directory can be flashed onto
+MSPM0G3507 LaunchPad using the steps mentioned below.
+
+Flashing
+========
+
+`UniFlash`_ is used to program the flash memory. Only bin loading is currently supported.
+Make sure to select the checkbox for binary loading.
+
```

Comment:
No `west flash` support? I believe having a way  to build and flash examples to a platform without relying on out of tree vendor tools is a basic requirement for a new platform being added. Ideally I will recommend to add OpenOCD support, but if that is not possible, is it possible to to invoke uniflash CLI from board.cmake?)

Reply:
Added comments to mention west flash. Open OCD support is there but pending a merge to be default